src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -312,7 +339,7 @@ internal async IAsyncEnumerable<Document> GetAllRegularAndSourceGeneratedDocumen
             return null;
 
         // Quick check first: if we already have created a SourceGeneratedDocument wrapper, we're good
-        if (_idToSourceGeneratedDocumentMap.TryGetValue(documentId, out var sourceGeneratedDocument))
+        if (_idToSourceGeneratedDocumentMap != null && _idToSourceGeneratedDocumentMap.TryGetValue(documentId, out var sourceGeneratedDocument))
```

Comment:
accessing out of lock.  not safe.

Reply:
Bah, this one was the ugly stepchild, and I only looked at the GetOrAdd usages, but yes, the TryGetValue calls need to be protected too. Added methods for both GetOrAdd and TryGetValue that encapsulate the locking.